### DISCUSSION TITLE
detect image manipulation by imageMagick

### DISCUSSION THREAD

**Discussion by anndawn** [ImageMagick/ImageMagick#6960] [Digital Image Processing] [answered]
Hi I have a set of documents of public interest that are processed with imageMagick and converted to pdfs. It seems the original images were also manipulated. How can I detect such manipulation, and is there a way to reverse it to a certain extent

**Comment by snibgo** [comment]
A digital image from a camera or scanner is always manipulated, because the raw data is not a simple colour image. It may be further manipulated to increase legibility and so on. And it might be manipulated to add forged signatures or whatever.

Detecting some manipulations may be possible, and might be reversible. Of course, if information has been removed, it can't be put back.

I suggest you post examples, with a description of what manipulations you want to detect and reverse.

**Reply by anndawn** [reply]
Thank you so much! It's forged stamps and intentional blurring we are trying to detect. Can I send the samples to your email directly? We would like to keep the documents confidential if possible.

**Reply by anndawn** [reply]
[These](https://drive.google.com/drive/folders/17vAe-h_IjVJJCQDpFYLS0PQ7STfGIiUa?usp=sharing) are some examples. Really appreciate your help

**Comment by urban-warrior** [comment]
ImageMagick generates a signature for each image, e.g., `identify -verbose | grep Sign`.  If the signature changes, the image pixels were modified.  You would need another method to determine if any of the image metadata was modified.  You can also compare two images and threshold the distortion.  If it exceeds some minimal value, the pixels were modified.

**Reply by anndawn** [reply]
Thank you so much! I don't have original file to compare with. Is there anyway I could contact you and send some sample of the document I talked about. Happy to pay for the support as well!

**Reply by urban-warrior** [reply]
We can only accept images if you post a link where we can download them.   Support is free through this forum, no payment required.

**Reply by anndawn** [reply]
Thanks. That's a good idea. [Here](https://drive.google.com/drive/folders/17vAe-h_IjVJJCQDpFYLS0PQ7STfGIiUa?usp=sharing) are some sample images that you can download

**Reply by urban-warrior** [reply]
Now what are you looking for?  Do you have your reference images that are converted to PDF and then you want to determine if anyone modified the PDF version?  Generally, to detect modifications, you would need an authoritative reference image to compare to the possibly manipulated version.  Describe in detail what you are trying to accomplish.

**Reply by anndawn** [reply]
Thanks! I don't have the original picture. These are what is provided by the government.  I could tell from the pdf metadata, the pdf is produced by Imagemagick.
With images where lines of the form seem to be breaking, is that some effect that could be produced by imagemagick?
With the blurred and pixelated pictures is there a way to tell if they are produced by imagemagick as well? Is there a way to trace some of the changes or manipulations of the document? Is there a way to reverse the changes to some extent? 
Also using photo forensic tools, I could tell with some certainty that some stamps are "fake". Does it leave any trace when they do these changes if they were using imagemagick? 
I guess in general if changes or modifications by imagemagick could be traced somehow? Thanks again!!!

**Comment by urban-warrior** [comment, marked as answer]
ImageMagick lacks built-in digital rights management for PDFs, making it challenging to authenticate whether a PDF has undergone manipulation. While some PDF solutions offer signing features for manipulation detection, ImageMagick does not currently support this capability. Essentially, ImageMagick takes the images you provide and encapsulates them within a PDF wrapper. If someone unpacks the image, makes modifications, and repackages it using ImageMagick, determining the alterations may prove difficult—unless utilizing specialized photo forensic tools, as you suggested.

**Reply by anndawn** [reply]
Thanks!! This is super helpful.
